rotkehlchen/premium/sync.py

```diff
@@ -115,6 +115,8 @@ def _sync_data_from_server_and_replace_local(self, perform_migrations: bool) ->
         try:
             self.data.decompress_and_decrypt_db(result['data'])
         except UnableToDecryptRemoteData as e:
+            self.data.db.disconnect()
```

Comment:
just to be explicit since you call it 2 times

```suggestion
            self.data.db.disconnect(conn_attribute='conn')
```

Also since this already raises `PremiumAuthenticationError` shouldn't the cleanup happen in `_abort_new_syncing_premium_user` which seems to be called wherever a `PremiumAuthenticationError` is handled?

Reply:
You're right.